payroll-processor-client/src/app/data/data-service.ts

```diff
@@ -53,11 +52,17 @@ export class DataService {
   }
 
   removeEmployee(id: string) {
-    this.employees.find(e => e.id === id).status = 'DISABLED';
```

Comment:
`find()` can return `undefined`, so I updated this to guard against attempting to access `.status` on `undefined`.

Same with the additional `.filter()` call below.

Reply:
Good catch, this is the type of stuff I am still learning about this language.

Reply:
Turning on the strict type checking will help too because the editor and compiler will error in situations like this.